src/porepy/models/solution_strategy.py

```diff
@@ -508,26 +516,78 @@ def check_convergence(
             # At least for the default direct solver, scipy.sparse.linalg.spsolve, no
             # error (but a warning) is raised for singular matrices, but a nan solution
             # is returned. We check for this.
-            diverged = bool(np.any(np.isnan(solution)))
+            diverged = bool(np.any(np.isnan(solution_increment)))
             converged: bool = not diverged
-            error: float = np.nan if diverged else 0.0
-            return error, converged, diverged
+            residual_error: float = np.nan if diverged else 0.0
+            increment_error: float = np.nan if diverged else 0.0
+            # TODO decide whether to omit logging of errors for linear problems
+            # return residual_error, increment_error, converged, diverged
```

Comment:
Note also that a unified way allows a simple access to residuals and increments (only true for all subproblems) for debugging in case a solution strategy of a linear problem does not generate the correct solution.